tests/pyret/tests/test-contracts.arr

```diff
@@ -319,52 +295,120 @@ check "should work in cases positions":
-  run-str("x :: NumNonPositive = 0.1") satisfies is-contract-error
-  run-str("x :: NumNonPositive = 0").success is true
+  run-str("x :: Exactnum = 5") is%(output) success
+  run-str("x :: Exactnum = ~5") is%(output) contract-error
+  run-str("x :: Roughnum = 5") is%(output) contract-error
+  run-str("x :: Roughnum = ~5") is%(output) success
+  run-str("x :: NumRational = ~5") is%(output) contract-error
+
+  run-str("x :: NumInteger = 5.1") is%(output) contract-error
+  run-str("x :: NumInteger = 5") is%(output) success
+
+  run-str("x :: NumRational = 5.1") is%(output) success
+  run-str("x :: NumRational = ~1") is%(output) contract-error
+
+  run-str("x :: NumPositive = ~5") is%(output) success
+  run-str("x :: NumPositive = ~-5") is%(output) contract-error
+  run-str("x :: NumPositive = 0.1") is%(output) success
+  run-str("x :: NumPositive = -0.1") is%(output) contract-error
+  run-str("x :: NumPositive = 0") is%(output) contract-error
+
+  run-str("x :: NumNegative = ~-5") is%(output) success
+  run-str("x :: NumNegative = ~5") is%(output) contract-error
+  run-str("x :: NumNegative = -0.1") is%(output) success
+  run-str("x :: NumNegative = 0.1") is%(output) contract-error
+  run-str("x :: NumNegative = 0") is%(output) contract-error
+
+  run-str("x :: NumNonNegative = ~5") is%(output) success
+  run-str("x :: NumNonNegative = ~-5") is%(output) contract-error
+  run-str("x :: NumNonNegative = 0.1") is%(output) success
+  run-str("x :: NumNonNegative = -0.1") is%(output) contract-error
+  run-str("x :: NumNonNegative = 0") is%(output) success
+
+  run-str("x :: NumNonPositive = ~-5") is%(output) success
+  run-str("x :: NumNonPositive = ~5") is%(output) contract-error
+  run-str("x :: NumNonPositive = -0.1") is%(output) success
+  run-str("x :: NumNonPositive = 0.1") is%(output) contract-error
+  run-str("x :: NumNonPositive = 0") is%(output) success
 end
 
 check "tuple contracts":
-  run-str("x :: {Number; String; {Number; Number; {String}}; String} = {4124; \"frwfq\"; {5123;531;{\"fqswf\"}}; \"fqwfq\"}").success is true
-  run-str("x :: {String; String} = {\"sewhwr\"; 4124}") satisfies is-contract-error
-  run-str("x :: {Number; String; {Number; Number; {String}}; String} = {4124; \"frwfq\"; {5123;531;{5351}}; \"fqwfq\"}") satisfies is-contract-error
-  run-str("x :: {Number; Number; {Number; String}} = {412; 5412; {412; \"fgwdef\"; 5135}}") satisfies is-contract-error
+  run-str("x :: {Number; String; {Number; Number; {String}}; String} = {4124; \"frwfq\"; {5123;531;{\"fqswf\"}}; \"fqwfq\"}") is%(output) success
+  run-str("x :: {String; String} = {\"sewhwr\"; 4124}") is%(output) contract-error
+  run-str("x :: {Number; String; {Number; Number; {String}}; String} = {4124; \"frwfq\"; {5123;531;{5351}}; \"fqwfq\"}") is%(output) contract-error
+  run-str("x :: {Number; Number; {Number; String}} = {412; 5412; {412; \"fgwdef\"; 5135}}") is%(output) contract-error
+end
+
+check "standalone contract statements":
```

Comment:
Good point.  It's not likely, since I'm walking over the lists of arguments and annotations in lockstep, rather than via fold2 or something (because I need to efficiently check whether they're the same length, too), but I'll add a test.